### PR TITLE
Use async file system access

### PR DESCRIPTION
**Pull request by a-hariti**

### CODE REVIEW

File: app/Controllers/Http/ExplorerController.js [hunk 1]
```diff
@@ -1099,15 +1103,15 @@ class ExplorerController {
...
+      if (await exists(path)) {
+        await writeFile(path, configStr, {encoding: 'utf8'});
         return true;
       }
     }
     return false;
   }
 
-  isDirectory = source => lstatSync(source).isDirectory();
+  isDirectory = async (source) => lstat(source).then(stat => stat.isDirectory());
```
Comment (0x7f) [inline app/Controllers/Http/ExplorerController.js:1114]: Right now the type of `isDirectry` is a promise. If you put an `await` before `async`, the variable is of type bool already and no one has to do an await on the isDirectory flag to read it.

**Review by 0x7f — APPROVED** [review]
LGTM. Just one minor comment. Didn't check whether really all sync calls are catched, though.

File: app/Controllers/Http/ExplorerController.js (hunk 1 above)
Comment (a-hariti) [reply app/Controllers/Http/ExplorerController.js:1114]: I agree that's a redundant `async`

**Comment by a-hariti** [comment]
@0x7f , a quick and dirty way to do it would be a `grep -R --exclude-dir node_modules/ --exclude-dir public Sync .` in the root of the repo.

File: app/Controllers/Http/ExplorerController.js (hunk 1 above)
Comment (0x7f) [reply app/Controllers/Http/ExplorerController.js:1114]: Forget what I said. I thought `isDirectory` is a local variable. But it is a short hand syntax for a member function. Then we should not add the `await` here to make `isDirectory` a bool, but it should stay a function. Whether you remove the `redundant` async or not, I don't care. From my point of view, this PR can be merged now :)

**Pull request merged**